### PR TITLE
🐛  use AZURE_VNET_NAME as vnetName in azure.json

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: To be consistent with AzureCluster.Spec.Vnet in templates.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Use AZURE_VNET_NAME as vnetName in azure.json
```

### CODE REVIEW

File: templates/cluster-template-system-assigned-identity.yaml [hunk 1]
```diff
@@ -200,7 +200,7 @@ spec:
             "securityGroupName": "${CLUSTER_NAME}-node-nsg",
             "location": "${AZURE_LOCATION}",
             "vmType": "vmss",
-            "vnetName": "${CLUSTER_NAME}-vnet",
+            "vnetName": "${CLUSTER_NAME}",
```
Comment (nader-ziada) [inline templates/cluster-template-system-assigned-identity.yaml:203]: should this be VNET_NAME?

File: templates/flavors/system-assigned-identity/patches/system-assigned-identity.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ spec:
             "securityGroupName": "${CLUSTER_NAME}-node-nsg",
             "location": "${AZURE_LOCATION}",
             "vmType": "vmss",
-            "vnetName": "${CLUSTER_NAME}-vnet",
+            "vnetName": "${CLUSTER_NAME}",
```
Comment (nader-ziada) [inline templates/flavors/system-assigned-identity/patches/system-assigned-identity.yaml:18]: should this be VNET_NAME?

Comment (CecileRobertMichon) [reply templates/flavors/system-assigned-identity/patches/system-assigned-identity.yaml:18]: oops yes typo

**Pull request merged**